ebean-api/src/main/java/io/ebean/bean/EntityBeanIntercept.java

```diff
@@ -239,6 +244,17 @@ public boolean isPartial() {
    * if any embedded beans are either new or dirty (and hence need saving).
    */
   public boolean isDirty() {
+    if (dirty) {
+      return true;
+    }
+    if (mutableHash != null) {
+      for (int i = 0; i < mutableHash.length; i++) {
+        if (mutableHash[i] != null && !mutableHash[i].isEqualToObject(owner._ebean_getField(i))) {
```

Comment:
if there are mutableHashes, I iterate over all of them and check if the hash thinks the value is dirty.
If so, the whole bean is considered as dirty. This is required for `DB.getBeanState(...).isDirty()`